### DISCUSSION TITLE
GitHub Actions workflow/CI checks for PRs

### DISCUSSION THREAD

**Discussion by issy** [Rapptz/discord.py#6911] [Ideas] [answered]
Would it be worth adding some CI workflows to add checks to PRs? I feel like it would help to ensure the code conforms to the same standard of quality across the repo, and would make sure no broken code is merged accidentally.
Of course, the maintainers of this project have generally been very hot on checking that the code is of a certain quality and doesn't break stuff, but it feels odd to have a project this large/popular and not have any CI.
I tried to find if a question like this had been asked in the past, but I couldn't find anything in Issues, PRs or Discussions. Thoughts?

**Comment by issy** [comment]
To extend this, tools I generally apply to this task are Black, Pylint, Flake8 and Isort. These can be configured to work nicely together, or you can pick 1/2 tools for the job. I generally quite like the Black style guidelines, and I think it makes for very neat and readable code.

**Comment by Rapptz** [comment, marked as answer]
I'm not really a fan of linters in Python since they're noisy. Likewise, I don't like black in its default configuration since it's too opinionated. This has been brought up before, see [here](https://github.com/Rapptz/discord.py/pull/6227) but currently I don't really want it. The only type of automation I'd want would probably be checking for extraneous whitespace but that's a different matter than what is proposed here.

**Reply by issy** [reply]
That makes some sense I suppose. IMO a well-configured linter doesn't work against you, but I suppose that's up to everyone's individual preference and style. Thanks for the answer
